cmd/collectors/restperf/restperf.go

```diff
@@ -715,6 +715,22 @@ func (r *RestPerf) PollData() (map[string]*matrix.Matrix, error) {
 
 	filter = append(filter, "counters.name="+strings.Join(metrics, "|"))
 
+	if isWorkloadDetailObject(r.Prop.Query) {
+		var workloadDetailFilter []string
+		resourceMap := r.Params.GetChildS("resource_map")
+		if resourceMap == nil {
+			return nil, errs.New(errs.ErrMissingParam, "resource_map")
+		}
+		for _, x := range resourceMap.GetChildren() {
```

Comment:
Removing line 719 and adding this line will reduce allocs.

`workloadDetailFilter := make([]string, 0, len(resourceMap.GetChildren()))`